src/coreclr/ToolBox/superpmi/mcs/verbjitflags.cpp

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include "standardpch.h"
+#include "verbjitflags.h"
+#include "methodcontext.h"
+#include "methodcontextiterator.h"
+#include "errorhandling.h"
+#include "corjitflags.h"
+
+int verbJitFlags::DoWork(const char* nameOfInput)
+{
+    MethodContextIterator mci;
+    if (!mci.Initialize(nameOfInput))
+        return -1;
+
+    LightWeightMap<unsigned long long, unsigned> flagMap;
+
+    while (mci.MoveNext())
+    {
+        MethodContext* mc = mci.Current();
+        CORJIT_FLAGS corJitFlags;
+        mc->repGetJitFlags(&corJitFlags, sizeof(corJitFlags));
+        unsigned long long rawFlags = corJitFlags.GetFlagsRaw();
+
+        int index = flagMap.GetIndex(rawFlags);
+        if (index == -1)
+        {
+            flagMap.Add(rawFlags, 1);
+        }
+        else
+        {
+            int oldVal = flagMap.GetItem(index);
+            flagMap.Update(index, oldVal + 1);
+        }
+    }
+
+    if (!mci.Destroy())
+        return -1;
+
+    printf("%16s,%8s, parsed\n", "bits", "count");
+
+    const unsigned int count = flagMap.GetCount();
+    unsigned long long* pFlag = flagMap.GetRawKeys();
+    for (unsigned int i = 0; i < count; i++)
+    {
+        const unsigned long long flag = *pFlag++;
+        const int index = flagMap.GetIndex(flag);
+
+        printf("%016llx,%8u", flag, flagMap.GetItem(index));
+
+        for (int flagBit = 63; flagBit >= 0; flagBit--)
```

Comment:
btw, this should probably be a spmi utility function that can also be used by `MethodContext::dmpGetJitFlags()`.

It also seems odd that you use `flagBit: 63->0 ; (flag >> flagBit) & 1ull` instead of `index: 0->63 ; flag & (1ull << index)`

Reply:
Good point, think I will adopt the model used by `SpmiDumpHelper::DumpCorInfoFlag`.

I was iterating backwards because I initially thought the higher-numbered flags were "more interesting."